src/middlewares/ValidateUser.js

```diff
@@ -1,25 +1,26 @@
 import models from '../api/models';
 import signupValidator from '../helpers/validators/signupValidator';
 import statusCode from '../helpers/constants/status.codes';
+import sendError from '../helpers/error.sender';
 
 const { User } = models;
 /**
  * @class
  */
 export default class {
   /**
-  * @param  {object} req The request object containing method, body, params, query,...ß
-  * @param  {object} res The response object for the request
-  * @param  {function} next The next middleware function in the request pipeline
-  * @returns {object} The response with status and data or error message
-  */
+   * @param  {object} req The request object containing method, body, params, query,...ß
+   * @param  {object} res The response object for the request
+   * @param  {function} next The next middleware function in the request pipeline
+   * @returns {object} The response with status and data or error message
+   */
   static async validateSignup(req, res, next) {
     const { email } = req.body;
     try {
       await signupValidator(req.body);
       const emailExists = await User.findByEmail(email);
-      if (emailExists) return res.status(statusCode.BAD_REQUEST).json({ message: ' The user already exists' });
-      return next();
+      if (emailExists) sendError(statusCode.BAD_REQUEST, res, 'email', ' The user already exists');
+      else return next();
     } catch (error) {
       if (error.isJoi) {
         const [err] = error.details;
```

Comment:
Can you  user  error.message.js helper  to send message please .to keep the consistency of the team error /message sending

Reply:
I'll do that. Actually, this file is supposed to be the ```autorization``` middleware that I'm wait ing for you. At the end I'm gonna remove it.

Reply:
alright,but the checkout is already done, you can check on it , while waiting  that my PR get ready to be merged

Reply:
cool